core/domain/exp_jobs_one_off.py

```diff
@@ -528,6 +528,123 @@ def reduce(key, values):
         yield (key, values)
 
 
+class ExplorationMathRichTextInfoModelGenerationOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Job that finds all the explorations with math rich text components and
+    creates a temporary storage model with all the information required for
+    generating math rich text component SVG images.
+    """
+
+    # A constant that will be yielded as a key by this job in the map function,
+    # When it finds an exploration with math rich text components without SVGs.
+    _SUCCESS_KEY = 'exploration-with-math-tags'
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        try:
+            exploration.validate()
+        except Exception as e:
+            logging.error(
+                'Exploration %s failed non-strict validation: %s' %
+                (item.id, e))
+            yield (
+                'validation_error',
+                'Exploration %s failed non-strict validation: %s' %
+                (item.id, e))
+            return
+        html_strings_in_exploration = ''
+        for state in exploration.states.values():
+            html_strings_in_exploration += (
+                ''.join(state.get_all_html_content_strings()))
+        list_of_latex_values_without_svg = (
+            html_validation_service.
+            get_latext_values_without_svg_from_html(
```

Comment:
latex is misspelled. Also, everywhere, please change "values" to "strings" where appropriate. Always use the most specific word possible.

Reply:
done